File: October Dusk 2020/Dhruv's - F.R.I.E.N.D.S/dhruv_friends.cpp
#include <bits/stdc++.h>
#define lli long long int
#define endl "\n"
#define MAX 1000005
#define MOD 1000000007
using namespace std;

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */ 
    string s1, s2,s3;
    cin>>s1>>s2>>s3;
    map <char,int >mp1;
    mp1['A'] = 0;
    mp1['S'] = 0;
    mp1['R'] = 0;
    if(s1[1] == '>')
        mp1[s1[0]]++;
    else
        mp1[s1[2]]++;
    if(s2[1] == '>')
        mp1[s2[0]]++;
    else
        mp1[s2[2]]++;
    if(s3[1] == '>')
        mp1[s3[0]]++;
    else
        mp1[s3[2]]++;
    
    if((mp1['A'] == mp1['S']) || (mp1['R'] == mp1['S']) || (mp1['A'] == mp1['R']))
        cout<<"Impossible";
    else{
        for(auto c: mp1)
        if(c.second == 0)
            cout<<c.first;
    for(auto c: mp1)
        if(c.second == 1)
            cout<<c.first;
    for(auto c: mp1)
        if(c.second == 2)
            cout<<c.first;
    }
    
    return 0;
}

File: October Dusk 2020/Dhruv's - F.R.I.E.N.D.S/dhruv_friends.py
flag = 0
p = {'A':0,'S':0,'R':0}
for _ in range(3):
    z = input()
    if z[1]=='>':
        p[z[0]]+=1
    else:
        p[z[2]]+=1
t = list(p.values())
if 0 not in t or 1 not in t or 2 not in t:
    print('Impossible')
    flag = 1
ans = [0,1,2]
if (not flag):
    for i in p:
        ans[p[i]]=i
    print(ans[0]+ans[1]+ans[2])
